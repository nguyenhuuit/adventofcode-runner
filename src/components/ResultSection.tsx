import React from 'react';

import { Box, Text } from 'ink';

import Spinner from '@components/Spinner';

interface ResultSectionProps {
  loading: boolean;
  answer: string;
  perfLog: string;
  output: string;
}

const ResultSection = ({ loading, answer, perfLog, output }: ResultSectionProps) => {
  return (
    <>
      <Box>
        <Text color="yellowBright">
          <Spinner type="earth" loading={loading} />
        </Text>
        <Text color="green">{`Result: `}</Text>
        <Text color="green" bold>
          {loading ? <Spinner type="simpleDots" loading={true} /> : answer}
        </Text>
        {answer && !loading && <Text>{`   ‚è± ${perfLog}`}</Text>}
      </Box>

      {!!output && (
        <>
          <Box height={1} />
          <Box borderTop flexDirection="column">
            <Text>{output}</Text>
          </Box>
        </>
      )}
    </>
  );
};

export default ResultSection;
