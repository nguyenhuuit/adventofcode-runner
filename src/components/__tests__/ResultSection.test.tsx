import { render } from '@testing-library/react';

import React from 'react';

import ResultSection from '../ResultSection';

interface MockBoxProps {
  children: React.ReactNode;
  [key: string]: unknown;
}

interface MockTextProps {
  children: React.ReactNode;
}

jest.mock('ink', () => ({
  Box: ({ children, ...props }: MockBoxProps) => (
    <div data-testid="box" {...props}>
      {children}
    </div>
  ),
  Text: ({ children }: MockTextProps) => <span data-testid="text">{children}</span>,
}));

jest.mock('../Spinner', () => ({
  __esModule: true,
  default: ({ loading }: { loading: boolean }) =>
    loading ? <div data-testid="spinner">Loading...</div> : null,
}));

describe('ResultSection', () => {
  it('renders correctly in loading state', () => {
    const { container } = render(<ResultSection loading={true} answer="" perfLog="" output="" />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with result', () => {
    const { container } = render(
      <ResultSection loading={false} answer="42" perfLog="0.5s" output="" />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with output', () => {
    const { container } = render(
      <ResultSection loading={false} answer="42" perfLog="0.5s" output="Test output" />
    );
    expect(container).toMatchSnapshot();
  });
});
