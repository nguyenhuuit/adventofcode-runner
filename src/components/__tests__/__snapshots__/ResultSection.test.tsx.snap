// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultSection renders correctly in loading state 1`] = `
<div>
  <div
    data-testid="box"
  >
    <span
      data-testid="text"
    >
      <div
        data-testid="spinner"
      >
        Loading...
      </div>
    </span>
    <span
      data-testid="text"
    >
      Result: 
    </span>
    <span
      data-testid="text"
    >
      <div
        data-testid="spinner"
      >
        Loading...
      </div>
    </span>
  </div>
</div>
`;

exports[`ResultSection renders correctly with output 1`] = `
<div>
  <div
    data-testid="box"
  >
    <span
      data-testid="text"
    />
    <span
      data-testid="text"
    >
      Result: 
    </span>
    <span
      data-testid="text"
    >
      42
    </span>
    <span
      data-testid="text"
    >
         ⏱ 0.5s
    </span>
  </div>
  <div
    data-testid="box"
    height="1"
  />
  <div
    data-testid="box"
    flexdirection="column"
  >
    <span
      data-testid="text"
    >
      Test output
    </span>
  </div>
</div>
`;

exports[`ResultSection renders correctly with result 1`] = `
<div>
  <div
    data-testid="box"
  >
    <span
      data-testid="text"
    />
    <span
      data-testid="text"
    >
      Result: 
    </span>
    <span
      data-testid="text"
    >
      42
    </span>
    <span
      data-testid="text"
    >
         ⏱ 0.5s
    </span>
  </div>
</div>
`;
